osquery/tables/system/windows/wmi_bios_info.cpp

```diff
@@ -155,52 +161,329 @@ Row getDellBiosInfo(const WmiResultItem& item) {
+              << key_name;
+
+      value.clear();
+    }
+
+    L_generateRow(key_name, std::move(value));
+  }
+
+  for (const auto& key_name : kStringVectorKeyNameList) {
+    std::vector<std::string> value_list;
+    auto status = wmi_item.GetVectorOfStrings(key_name, value_list);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+
+      value_list.clear();
+    }
+
+    std::string buffer = "{";
+    for (const auto& value : value_list) {
+      if (buffer.size() > 1) {
+        buffer += ", ";
+      }
+
+      buffer += value;
+    }
+
+    buffer += "}";
+
+    L_generateRow(key_name, std::move(buffer));
+  }
+
+  for (const auto& key_name : kUInt16KeyNameList) {
+    long value{};
+    auto status = wmi_item.GetLong(key_name, value);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+      value = 0;
+    }
+
+    L_generateRow(key_name, std::to_string(value));
+  }
+
+  for (const auto& key_name : kUInt16VectorKeyNameList) {
+    std::vector<long> value_list;
+    auto status = wmi_item.GetVectorOfLongs(key_name, value_list);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+      value_list.clear();
+    }
+
+    std::string buffer = "{";
+    for (const auto& value : value_list) {
+      if (buffer.size() > 1) {
+        buffer += ", ";
+      }
+
+      buffer += std::to_string(value);
+    }
+
+    buffer += "}";
+
+    L_generateRow(key_name, std::move(buffer));
+  }
+
+  for (const auto& key_name : kUInt8KeyNameList) {
+    unsigned char value{};
+    auto status = wmi_item.GetUChar(key_name, value);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+      value = 0;
+    }
+
+    L_generateRow(key_name, std::to_string(value));
+  }
+
+  for (const auto& key_name : kBooleanKeyNameList) {
+    bool value{};
+    auto status = wmi_item.GetBool(key_name, value);
+    if (!status.ok()) {
+      VLOG(1) << "Failed to copy the following Win32_BIOS WMI column value: "
+              << key_name;
+      value = false;
+    }
+
+    L_generateRow(key_name, value ? "true" : "false");
+  }
+
+  for (const auto& key_name : kDateKeyNameList) {
+    SYSTEMTIME system_time{};
+
+    {
+      FILETIME value{};
+      auto status = wmi_item.GetDateTime(key_name, false, value);
+      if (status.ok() && !FileTimeToSystemTime(&value, &system_time)) {
+        status = Status::failure("Failed to convert the FILETIME structure");
+      }
+
+      if (!status.ok()) {
+        system_time = {};
+        VLOG(1) << "Failed to acquire the following column value: " << key_name
+                << ". " << status.getMessage();
+      }
+    }
+
+    L_generateRow(key_name, getISODateFromSystemTime(system_time));
```

Comment:
We do have a `osquery::filetimeToUnixtime` so maybe the question is whether we'd rather these in unix time, or ISO dates. 

My bias is usually to consistency, which is unix dates. But 🤷